integration_tests/src/template.rs

```diff
@@ -11,17 +12,93 @@ use minotari_app_grpc::tari_rpc::{
     WasmInfo,
 };
 use tari_dan_engine::wasm::compile::compile_template;
-use tari_engine_types::{hashing::template_hasher32, TemplateAddress};
+use tari_engine_types::{
+    commit_result::TransactionResult,
+    hashing::template_hasher32,
+    substate::SubstateId,
+    TemplateAddress,
+};
 use tari_template_lib::Hash;
+use tari_wallet_daemon_client::{
+    types::{PublishTemplateRequest, TransactionWaitResultRequest},
+    ComponentAddressOrName,
+};
 
-use crate::TariWorld;
+use crate::{wallet_daemon_cli::get_auth_wallet_daemon_client, TariWorld};
 
 #[derive(Debug)]
 pub struct RegisteredTemplate {
     pub name: String,
     pub address: TemplateAddress,
 }
 
+pub async fn publish_template(
+    world: &mut TariWorld,
+    wallet_daemon_name: String,
+    account_name: String,
+    template_name: String,
+) -> anyhow::Result<TemplateAddress> {
+    // compile and load wasm
+    compile_wasm_template(template_name.clone())?;
+    let wasm_file_path = get_template_wasm_path(template_name.clone());
+    let wasm_binary = tokio::fs::read(&wasm_file_path).await?;
+
+    // send publish template request
+    let mut client = get_auth_wallet_daemon_client(world, &wallet_daemon_name).await;
+    let response = client
+        .publish_template(PublishTemplateRequest {
+            binary: wasm_binary,
+            fee_account: Some(ComponentAddressOrName::Name(account_name)),
+            max_fee: 1_000_000,
+            detect_inputs: true,
+            dry_run: false,
+        })
+        .await?;
+
+    let tx_resp = client
+        .wait_transaction_result(TransactionWaitResultRequest {
+            transaction_id: response.transaction_id,
+            timeout_secs: Some(60 * 5),
+        })
+        .await?;
+
+    if tx_resp.timed_out {
+        return Err(anyhow!(format!("Transaction timed out: {}", response.transaction_id)));
+    }
+
+    let finalize_result = tx_resp.result.ok_or(anyhow!(format!(
+        "Missing transaction result: {}",
+        response.transaction_id
+    )))?;
+    if !matches!(finalize_result.result, TransactionResult::Accept(_)) {
+        let error_status = match finalize_result.result {
+            TransactionResult::AcceptFeeRejectRest(_, reason) | TransactionResult::Reject(reason) => {
+                format!("Status: {}, Reason: {}", tx_resp.status, reason)
+            },
+            TransactionResult::Accept(_) => String::new(), // does not happen here
+        };
+        return Err(anyhow!(format!(
+            "Invalid transaction {}: {error_status:?}",
+            response.transaction_id
+        )));
+    }
```

Comment:
```suggestion
    if let Some(reason) = finalize_result.result.full_reject() {
        return Err(anyhow!(format!(
            "Invalid transaction {}: Status: {}, Reason: {}", 
            response.transaction_id, tx_resp.status, reason
        )));
    }
```

Reply:
Yeah the names are really bad - maybe we can make them fee_transaction_accept (prev. full_accept), accept, fee_transaction_reject (prev. full_reject), reject. These always confuse me and have caused bugs (even found one related to that today)